src/Serilog/Context/LogContext.cs

```diff
@@ -12,9 +12,16 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+
 #if LOGCONTEXT
```

Comment:
Ah, that's a shame. What do you think (+ @merbla @khellang) about using `[ThreadStatic]` for this on 5.1?

There are two benefits:

1. `LogContext` is a pretty fundamental API, and it is still useful even without async support (better than nothing, if you're targeting very limited surface area!), but more importantly:
2. Libraries may _build_ against the 5.1 surface area, but then be installed in applications that do in fact target 4.5+ or 5.4+ - in these cases, the binary used at runtime would be the version with full `LogContext` support ([bait-and-switch](http://log.paulbetts.org/the-bait-and-switch-pcl-trick/))

Worth a shot?